test/test_cuts.py

```diff
@@ -0,0 +1,241 @@
+mass1 = list(np.random.random(n_templates - 1) * 100) + [59]
+mass2 = list(np.random.random(n_templates - 1) * 100) + [10]
+
+# Assert mass1 > mass2
+template_dset['mass1'] = np.maximum(mass1, mass2)
+template_dset['mass2'] = np.minimum(mass1, mass2)
+
+# Spins must be in the range -1, 1, (but not exactly 1):
+def make_random_spins(n_templates):
+    spins = np.random.normal(size=n_templates, scale=0.4)
+    spins = np.maximum(spins, -0.998)
+    spins = np.minimum(spins, 0.998)
+    return spins
+
+template_dset['spin1z'] = make_random_spins(n_templates)
+template_dset['spin2z'] = make_random_spins(n_templates)
+
+template_dset['template_duration'] = np.random.random(n_templates) * 100
+
+
+class CutsTest(unittest.TestCase):
+    def setUp(self):
+        # Set up some things we will want to use in the tests:
+        self.maxDiff = None
+
+# Set up values to be tested:
+test_cut_output = []
+
+
+# Lower limits on a trigger parameter we can read directly
+def snr_lower_test(trigs, trig_idx, **kwargs):
+    return all(trigs['snr'][trig_idx] > 4)
+
+
+test_cut_output.append((['--trigger-cuts', 'snr:4:lower'],
+                        snr_lower_test,
+                        'snr_cut_lower'))
+
+
+# Upper limits on a derived trigger parameter
+def newsnr_sgveto_upper_test(trigs, trig_idx, **kwargs):
+    rwsnr = ranking.get_newsnr_sgveto(trigs)
+    return all(rwsnr[trig_idx] < 10)
+
+test_cut_output.append((['--trigger-cuts', 'newsnr_sgveto:10:upper'],
+                        newsnr_sgveto_upper_test,
+                        'nsnr_sgveto_cut_upper'))
+
+# Lower limit on directly-read template bank parameter
+def template_duration_test(temps, temp_idx, **kwargs):
+    return all(temps['template_duration'][temp_idx] > 10)
+
+test_cut_output.append((['--template-cuts',
+                         'template_duration:10:lower'],
+                        template_duration_test,
+                        'template_duration_lower'))
+
+# make sure the "or equal to" is working properly
+def mass1_ge_test(temps, temp_idx, **kwargs):
+    return any(temps['mass1'][temp_idx] == 59)
+
+test_cut_output.append((['--template-cuts',
+                         'mass1:59:lower_inc'],
+                        mass1_ge_test,
+                        'mass1_orequal'))
+
+# Upper and lower limits on a derived parameter from the bank
+def chi_eff_upper_lower_test(temps, temp_idx, **kwargs):
+    chi_eff = bank_conversions.get_bank_property('chi_eff', temps, temp_idx)
+    return all(np.logical_and(chi_eff > -0.5,
+                              chi_eff <= 0.8))
+
+
+test_cut_output.append((['--template-cuts', 'chi_eff:0.8:upper',
+                         'chi_eff:-0.5:lower_inc'],
+                         chi_eff_upper_lower_test,
+                         'chi_eff_cut_upper_lower'))
+
+
+# FIXME: Once we can fake the statistic files, create a test
+# for template fits cuts
+
+# Dynamically add value tests for the parser
+for test_values in test_cut_output:
```

Comment:
I don't understand the motivation for doing this? It confused me reading over this that the test objects don't have any contents. But then I see that it's added here. What is the reason why we can't just define the tests as methods of the class (even if they just call other methods/utility functions with varying inputs).

Reply:
I was wanting to make it as easy as possible to add new tests into the suite, and this definition seemed easiest.

I tried doing the test definitions as methods of the class with variable input, but I seemed to be just getting the same test running multiple times, and this didn't

Doing it this way, we can add in a new test using only the inputs, required outputs and a name for the test. It is stranger in this case as we are defining the required outputs by a function that must return `True`, but it remains consistent with the other tests in this module.